api/v1alpha2/bundledeployment_types.go

```diff
@@ -69,8 +69,42 @@ type BundleDeploymentSpec struct {
 	//
 	// config is provisioner specific configurations
 	Config runtime.RawExtension `json:"config,omitempty"`
+
+	//+kubebuilder:Optional
+	// Preflight defines the strategy for the preflight checks (i.e. as of now, the CRD upgrade safety checks) to be enabled or disabled when attempting to install the BundleDeployment.
+	Preflight *PreflightConfig `json:"preflight,omitempty"`
+}
+
+// PreflightConfig holds the config for the preflight checks. Currently, this is implemented for the
+// CRDUpgradeSafety preflight check and can be extended in the future to add other preflight checks.
```

Comment:
```suggestion
// PreflightConfig holds the config for the preflight checks.
```

This comment makes it into the CRD API docs. The "Currently ...." bit is developer doc that shouldn't show up there.